File: translate_quotes.py
import re
import sys
import asyncio
from googletrans import Translator

async def translate_quotes():
    with open(r'C:\Users\PC\Desktop\antigravy painto gain\lib\backend\quotes_repository.dart', 'r', encoding='utf-8') as f:
        content = f.read()

    translator = Translator()

    # Find all Quote objects
    # We will look for localizedText: { ... }
    # Inside, we find 'en': '...' and use it to translate.
    
    # We need to find all blocks of localizedText
    pattern = re.compile(r"(localizedText:\s*\{)(.*?)(\},)", re.DOTALL)
    
    async def process_block(match):
        prefix = match.group(1)
        inner = match.group(2)
        suffix = match.group(3)
        
        # If already translated, skip
        if "'it':" in inner and "'pt':" in inner:
            return match.group(0)
            
        # Find English text
        en_match = re.search(r"'en':\s*'((?:\\'|[^'])*)'", inner)
        if not en_match:
            return match.group(0)
            
        en_text = en_match.group(1).replace("\\'", "'")
        
        try:
            # Translate to Italian and Portuguese
            it_trans = await translator.translate(en_text, dest='it')
            pt_trans = await translator.translate(en_text, dest='pt')
            
            it_text = it_trans.text.replace("'", "\\'")
            pt_text = pt_trans.text.replace("'", "\\'")
            
            # Append to inner
            new_inner = inner + f"      'it': '{it_text}',\n      'pt': '{pt_text}',\n"
            print(f"Translated: {en_text[:30]}...")
            return prefix + new_inner + suffix
        except Exception as e:
            print(f"Error translating: {e}")
            return match.group(0)

    # We can't use async re.sub, so we'll do it manually
    pos = 0
    new_content = ""
    for match in pattern.finditer(content):
        new_content += content[pos:match.start()]
        new_content += await process_block(match)
        pos = match.end()
    
    new_content += content[pos:]

    with open(r'C:\Users\PC\Desktop\antigravy painto gain\lib\backend\quotes_repository.dart', 'w', encoding='utf-8') as f:
        f.write(new_content)
        
    print("Translation completed successfully.")

if __name__ == '__main__':
    asyncio.run(translate_quotes())

File: translate_quotes.dart
import 'dart:io';
import 'package:translator/translator.dart';

void main() async {
  final file = File('lib/backend/quotes_repository.dart');
  var content = await file.readAsString();
  final translator = GoogleTranslator();

  // Pattern to match localizedText maps
  // Format: localizedText: {\n      'tr': '...', \n      'en': '...', ...\n    },
  final regex = RegExp(r"(localizedText:\s*\{)(.*?)(\},)", dotAll: true);
  
  final matches = regex.allMatches(content).toList();
  var newContent = content;

  for (var match in matches) {
    final prefix = match.group(1)!;
    final inner = match.group(2)!;
    final suffix = match.group(3)!;

    if (inner.contains("'it':") && inner.contains("'pt':")) {
      continue;
    }

    final enRegex = RegExp(r"'en':\s*'((?:\\'|[^'])*)'");
    final enMatch = enRegex.firstMatch(inner);
    
    if (enMatch != null) {
      final enText = enMatch.group(1)!.replaceAll("\\'", "'");
      try {
        final itTrans = await translator.translate(enText, to: 'it');
        final ptTrans = await translator.translate(enText, to: 'pt');
        
        final itText = itTrans.text.replaceAll("'", "\\'");
        final ptText = ptTrans.text.replaceAll("'", "\\'");
        
        final newInner = inner + "      'it': '$itText',\n      'pt': '$ptText',\n";
        
        final originalMatch = match.group(0)!;
        final replacement = prefix + newInner + suffix;
        newContent = newContent.replaceFirst(originalMatch, replacement);
        print("Translated: ${enText.substring(0, enText.length > 30 ? 30 : enText.length)}...");
      } catch (e) {
        print("Error: $e");
      }
    }
  }

  await file.writeAsString(newContent);
  print("Done!");
}
